### DISCUSSION TITLE
Using a GPU for LeanCopilot

### DISCUSSION THREAD

**Discussion by realharryhero** [lean-dojo/LeanCopilot#76] [Q&A] [answered]
Hello,

I have a NVIDIA GeForce RTX 3090 that I would like to use with LeanCopilot. Is there any way to do so? For example, when using lake exe LeanCopilot/download, I see the following text:

```
ℹ [3/18] Replayed LeanCopilot/libopenblas
info: Cloning OpenBLAS from https://github.com/OpenMathLib/OpenBLAS
info: Building OpenBLAS with `make NO_LAPACK=1 NO_FORTRAN=1 -j8`
ℹ [4/18] Replayed LeanCopilot/libctranslate2
info: Cloning CTranslate2 from https://github.com/OpenNMT/CTranslate2
info: Configuring CTranslate2 with `cmake -DBUILD_CLI=OFF -DOPENMP_RUNTIME=NONE -DWITH_DNNL=OFF -DWITH_MKL=OFF -DWITH_ACCELERATE=OFF -DWITH_OPENBLAS=ON -DOPENBLAS_INCLUDE_DIR=../../OpenBLAS -DOPENBLAS_LIBRARY=../../OpenBLAS/libopenblas.so -DWITH_CUDA=OFF -DWITH_CUDNN=OFF ..`
```

I assume I'd like to set -DWITH_CUDA=ON and -DWITH_CUDNN=ON; how could I do so?

**Comment by yangky11** [comment, marked as answer]
We use `which nvcc` to detect whether CUDA is available. Do you have `nvcc` installed?

**Comment by realharryhero** [comment]
When I used the GPU at that time, no. I'll set this as closed for now since that's probably the issue, but I can't use the GPU (I broke it; don't install CUDA 12.1 with kernel version 6.5, if anyone's having the same issue) recently. Thank you!

**Comment by vanessalama09** [comment]
@yangky11  I'm also trying to build Lean Copilot on NVIDIA cuda-compatible GPU. I was able to build locally, but on the GPU, it's throwing an error at the same stage as above. A little about the machine:
- linux x86_64
- NVIDIA A100 GPUs

The build error:

```
ℹ [3/1126] Replayed LeanCopilot/libopenblas
info: Cloning OpenBLAS from https://github.com/OpenMathLib/OpenBLAS
info: Building OpenBLAS with `make NO_LAPACK=1 NO_FORTRAN=1 -j32`
✖ [4/1126] Building LeanCopilot/libctranslate2
info: Cloning CTranslate2 from https://github.com/OpenNMT/CTranslate2
error: no such file or directory (error code: 2)
  file: ././.lake/packages/LeanCopilot/.lake/build/lib/libctranslate2.so
Some builds logged failures:
- LeanCopilot/libctranslate2
error: build failed
```

A little help would be much appreciated!
